joint_limits/include/joint_limits/joint_limits_helpers.hpp

```diff
@@ -0,0 +1,87 @@
+// Copyright 2024 PAL Robotics S.L.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// \author Adrià Roig Moreno
+
+#ifndef JOINT_LIMITS__JOINT_LIMITS_HELPERS_HPP_
+#define JOINT_LIMITS__JOINT_LIMITS_HELPERS_HPP_
+
+#include <optional>
+#include <string>
+#include <utility>
+#include "joint_limits/joint_limits.hpp"
+
+namespace joint_limits
+{
+
+/**
+ * @brief Checks if a value is limited by the given limits.
+ * @param value The value to check.
+ * @param min The minimum limit.
+ * @param max The maximum limit.
+ * @return True if the value is limited, false otherwise.
+ */
+bool is_limited(double value, double min, double max);
+
+/**
+ * @brief Computes the position limits based on the velocity and acceleration limits.
+ * @param limits The joint limits.
+ * @param act_vel The actual velocity of the joint.
+ * @param prev_command_pos The previous commanded position of the joint.
+ * @param dt The time step.
+ * @return The position limits.
+ */
+std::pair<double, double> compute_position_limits(
```

Comment:
I have a suggestion from a code reader's point of view. What do you think about creating a struct with the min and max members instead? Using std::pair results later in the use of `first` and `second` accessors, which does not describe the purpose of the variable.

Reply:
Hello!

As these are helper methods alone. I want to keep them simple. Making a new struct and passing them around is something I wanted to avoid. That's the main reason I avoided using such new structs

Reply:
Gotta do better in the documentation at least then sir.

`* @return The position limits.` Which one is which?

Reply:
That being said if we wanted to go fully proper, I'd say a strongly typed "VelocityLimits" and "AccelerationLimits" would not be too bad either so it is easier to track down if they are used in the right places...

Reply:
I've created an issue for this thing. I've also updated the documentation mentioning which one corresponds to what